### PR TITLE
Fix issue of metadata only registred to one nacos namespace 

### PR DESCRIPTION
**Pull request by namelessssssssssss**
## What is the purpose of the change
Fix issue https://github.com/apache/dubbo/issues/13371

### CODE REVIEW

**Comment by namelessssssssssss** [comment]
@AlbumenJ PTAL :)

**Review by AlbumenJ — APPROVED** [review]

File: dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/MetadataReportFactory.java [hunk 1]
```diff
@@ -34,4 +34,9 @@ public interface MetadataReportFactory {
     MetadataReport getMetadataReport(URL url);
 
     default void destroy() {}
+
+    @Adaptive({PROTOCOL_KEY})
+    default String relatedRegistryIdSuffix(URL url) {
+        return "";
+    }
```
Comment (chickenlj) [inline dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/MetadataReportFactory.java:41]: This method here makes the interface definition hard to understand.

Factory is only responsible for creating new instances.

**Review by chickenlj — CHANGES_REQUESTED** [review]

File: dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/MetadataReportInstance.java [hunk 2]
```diff
@@ -110,6 +110,8 @@ private void init(MetadataReportConfig config, MetadataReportFactory metadataRep
         // exist."));
         MetadataReport metadataReport = metadataReportFactory.getMetadataReport(url);
         if (metadataReport != null) {
+            System.out.println("MetadataReportFactory:"+ metadataReportFactory.getClass()+",MetadataReport:"+metadataReport+" url:"+url+" ,relatedRegistryId:"+metadataReport.relatedRegistryIdSuffix(url));
```
Comment (AlbumenJ) [inline dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/MetadataReportInstance.java:113]: Sout is not allow

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/MetadataReportInstance.java (hunk 2 above)
Comment (namelessssssssssss) [reply dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/report/MetadataReportInstance.java:113]: Just for temporary test,I will remove it later

**Comment by AlbumenJ** [comment]
Please fix on 3.2.x :)

**Comment by namelessssssssssss** [comment]
This issue fixed on 3.2.x

**Pull request closed without merging**